Bottom: 8374706fe50b87ba0cf02af4baf3d04343929d64
Top:    30ff6921cec3119658fd06704428e569b8a5676f
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2012-12-18 02:22:40 +0000

drc: HACK HACK HACK


---

diff --git a/src/find.c b/src/find.c
index 3fbd646..1d90bc1 100644
--- a/src/find.c
+++ b/src/find.c
@@ -2507,7 +2507,12 @@ IsPolygonInPolygon (PolygonType *P1, PolygonType *P2)
 
               line.Point1.X = v->point[0];
               line.Point1.Y = v->point[1];
-              line.Thickness = 2 * Bloat;
+              line.Thickness = Bloat;
+              /* Another Bloat is added by IsLineInPolygon, making the correct
+               * 2x Bloat. Ideally we would change it there, but doing so
+               * breaks some other DRC checks which rely on the broken behaviour
+               * in IsLineInPolygon.
+               */
               line.Clearance = 0;
               line.Flags = NoFlags ();
               for (v = v->next; v != &c->head; v = v->next)
